src/main/scala/rocket/RocketCore.scala

```diff
@@ -313,7 +327,8 @@ class Rocket(tile: RocketTile)(implicit p: Parameters) extends CoreModule()(p)
     id_ctrl.rocc && csr.io.decode(0).rocc_illegal ||
     id_ctrl.scie && !(id_scie_decoder.unpipelined || id_scie_decoder.pipelined) ||
     id_csr_en && (csr.io.decode(0).read_illegal || !id_csr_ren && csr.io.decode(0).write_illegal) ||
-    !ibuf.io.inst(0).bits.rvc && (id_system_insn && csr.io.decode(0).system_illegal)
+    !ibuf.io.inst(0).bits.rvc && (id_system_insn && csr.io.decode(0).system_illegal) ||
+    id_ctrl.zkn && id_ctrl.alu_fn(ZKN.FN_Len-1,0) === ZKN.FN_AES_KS1 && id_inst(0)(23,20) > 0xA.U(4.W)
```

Comment:
`id_ctrl.alu_fn(ZKN.FN_Len-1,0) === ZKN.FN_AES_KS1` needs additional comparing logic, which should be better to wire a `isAESKS1` signal out form the decoder.
you can add a TODO here.

Reply:
I'll redesign the uop into control signals just as ABLU had done.

Reply:
Now it uses one signal from fn: `id_ctrl.alu_fn(15)`.

The comparing logic for `id_inst(0)(23,20) > 0xA.U(4.W)` could not be avoided, though.